### DISCUSSION TITLE
Why is wxt/storage async when localStorage is not?

### DISCUSSION THREAD

**Discussion by BingeCode** [wxt-dev/wxt#1066] [Q&A] [answered]
What are the benefits or constraints that lead to this design decision?

Isn't it much simpler to use sync operations when dealing with localStorage?

**Comment by aklinker1** [comment]
It doesn't use `localStorage` internally, it uses [`chrome.storage`](https://developer.chrome.com/docs/extensions/reference/api/storage), which is async.

**Reply by BingeCode** [reply]
That is interesting. What are the benefits of using chrome.storage instead of localStorage? And how does wxt/storage make sure it works in Firefox?

**Reply by aklinker1** [reply, marked as answer]
Firefox supports the `chrome` global, so it just works.

As for the differences, read the ["concepts and usages" section](https://developer.chrome.com/docs/extensions/reference/api/storage#usage) from the previous link.

But TLDR `localStorage` wasn't designed for extensions and doesn't work how you think it should.
